cloud/kubernetes/helm/yugabyte/templates/service.yaml

```diff
@@ -176,20 +181,24 @@ spec:
           - "/home/yugabyte/bin/yb-master"
           - "--fs_data_dirs={{range $index := until (int ($root.Values.persistentVolume.count))}}{{if ne $index 0}},{{end}}/mnt/disk{{ $index }}{{end}}"
           - "--rpc_bind_addresses=$(POD_IP)"
-          - "--server_broadcast_addresses=$(HOSTNAME).yb-masters:7100"
-          - "--master_addresses={{range $index := until (int ($root.Values.replicas.master))}}{{if ne $index 0}},{{end}}yb-master-{{ $index }}.yb-masters:7100{{end}}"
+          - "--server_broadcast_addresses=$(HOSTNAME).yb-masters.$(NAMESPACE):7100"
+          - "--master_addresses={{range $index := until (int ($root.Values.replicas.master))}}{{if ne $index 0}},{{end}}yb-master-{{ $index }}.yb-masters.$(NAMESPACE):7100{{end}}"
           - "--master_replication_factor={{ $root.Values.replicas.master }}"
           - "--metric_node_name=$(HOSTNAME)"
+          - "--memory_limit_hard_bytes={{ $root.Values.resource.master.limits.memory | regexFind "\\d+" | mul 1024 | mul 1024 | mul 870 }}"
```

Comment:
So we are assuming that "limits.memory" is in Gi, and does the regexFind drop the Gi part?

Reply:
1) would be nice if there was a way to add comments to explain the math, and the above assumption.

2) The 870 would later become 1024 I suppose, once the enhancement for https://github.com/YugaByte/yugabyte-db/issues/628 lands, right?

Reply:
Yes, we are assuming the memory is in GB (since that is how it is driven from YW). 
1) I'll add comments and create a separate PR.
2) Correct, currently the helm templates do not support floating point calculations and that is why I had to do it this way (could add a divide as well, but seemed like overkill)